docs/en/operations/settings/settings.md

```diff
@@ -3310,6 +3310,15 @@ SELECT
 FROM fuse_tbl
 ```
 
+## optimize_rewrite_aggregate_function_with_if
+
+Rewrite aggregate functions with if expression as argument when logically equivalent.
+For example, `avg(if(cond, col, null))` can be rewritten to `avgIf(cond, col)`. It may improve performance.
```

Comment:
As far as I understand it works like that:
```suggestion
For example, `avg(if(cond, col, null))` can be rewritten to `avgIfOrNull(cond, col)`. It may improve performance.
```

Reply:
Yes, thanks

```
EXPLAIN QUERY TREE dump_ast = 1
SELECT avg(if((x % 2) = 0, x, NULL))
FROM
(
    SELECT number AS x
    FROM numbers(10)
)

Query id: 6bae269d-eed0-421f-8a57-a9546abead14

┌─explain────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ QUERY id: 0                                                                                                │
│   PROJECTION COLUMNS                                                                                       │
│     avg(if(equals(modulo(x, 2), 0), x, NULL)) Nullable(Float64)                                            │
│   PROJECTION                                                                                               │
│     LIST id: 1, nodes: 1                                                                                   │
│       FUNCTION id: 2, function_name: avgOrNullIf, function_type: aggregate, result_type: Nullable(Float64) │
│         ARGUMENTS                                                                                          │
│           LIST id: 3, nodes: 2                                                                             │
│             COLUMN id: 4, column_name: x, result_type: UInt64, source_id: 5                                │
│             FUNCTION id: 6, function_name: equals, function_type: ordinary, result_type: UInt8             │
│               ARGUMENTS                                                                                    │
│                 LIST id: 7, nodes: 2                                                                       │
│                   FUNCTION id: 8, function_name: modulo, function_type: ordinary, result_type: UInt8       │
│                     ARGUMENTS                                                                              │
│                       LIST id: 9, nodes: 2                                                                 │
│                         COLUMN id: 4, column_name: x, result_type: UInt64, source_id: 5                    │
│                         CONSTANT id: 10, constant_value: UInt64_2, constant_value_type: UInt8              │
│                   CONSTANT id: 11, constant_value: UInt64_0, constant_value_type: UInt8                    │
│   JOIN TREE                                                                                                │
│     QUERY id: 5, is_subquery: 1                                                                            │
│       PROJECTION COLUMNS                                                                                   │
│         x UInt64                                                                                           │
│       PROJECTION                                                                                           │
│         LIST id: 12, nodes: 1                                                                              │
│           COLUMN id: 13, column_name: number, result_type: UInt64, source_id: 14                           │
│       JOIN TREE                                                                                            │
│         TABLE_FUNCTION id: 14, table_function_name: numbers                                                │
│           ARGUMENTS                                                                                        │
│             LIST id: 15, nodes: 1                                                                          │
│               CONSTANT id: 16, constant_value: UInt64_10, constant_value_type: UInt8                       │
│                                                                                                            │
│ SELECT avgOrNullIf(x, (x % 2) = 0)                                                                         │
│ FROM                                                                                                       │
│ (                                                                                                          │
│     SELECT number                                                                                          │
│     FROM numbers(10)                                                                                       │
│ )                                                                                                          │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

37 rows in set. Elapsed: 0.001 sec. 
```